### ISSUE TITLE
Add auth indication

### ISSUE THREAD

**Issue by MV-GH** [LemmyNet/lemmy-js-client#199]
With 0.19 now its not obvious which endpoints need authentication and which not. Maybe we should annotate that information in the JSDoc. This is especially useful for my openapi spec. As I can parse this JSDoc and auto add which endpoints need auth or not.

**Comment by dessalines** [comment]
We could annotate the http methods in some way, but it wouldn't be a good idea to do so for the auto-generated request objects.

I'm not sure what typedoc label would work best tho: https://tsdoc.org/pages/tags/public/

**Comment by MV-GH** [comment]
Yes, I meant the http routes. It doesn't have to be a real annotation. Could just be `@Authenticed`.But what should we do for methods that have different response based on auth. see GET /site

**Comment by MV-GH** [comment]
Yeah for those routes without auth we simply don't add a annotation.

So `@Auth` for full authenticated routes. So `@AuthElevated` or something for routes where auth elevates the response? 

It should also be documented what each annotation means in that http file

**Comment by dessalines** [comment]
I'd be good with either of those.

**Comment by NatoBoram** [comment]
It doesn't necessarily need to be an annotation, it could also be a sentence. Something like *Requires authentication.* or *Requires moderator privileges.* or *Requires administrator privileges.*.

**Comment by MV-GH** [comment]
Well it doesn't matter to me as long as it can easily be parsed. (The above solution doesn't seem like that though)

**Comment by Nutomic** [comment]
In the Lemmy code you can see that an endpoint requires auth if the handler method takes param [`LocalUserView`](https://github.com/LemmyNet/lemmy/blob/main/crates/api/src/comment/like.rs#L27). If auth is optional the param is [`Option<LocalUserView>`](https://github.com/LemmyNet/lemmy/blob/main/crates/api_crud/src/comment/read.rs#L16). If there is no `LocalUserView` param then auth makes no difference, but I dont think there is such an endpoint (except RSS feeds etc).

So it would be great if this could be generated from the Rust code, but its probably not possible due to lack of tooling.

**Comment by MV-GH** [comment]
I guess this has been achieved with the openapi spec? All though that misses the full semantics of the optional auth endpoints, where Auth is just an enrichment.

**Comment by dessalines** [comment]
Yep I did go through and add the `@Security` tags on the specific endpoints where its optional or required. 

I think just adding this is how you'd specify that security is optional on an endpoint:

```ts
@Security({})
@Security("bearerAuth")
```
